### ISSUE TITLE
[Bug] Stashing single or individual files doesn't work correctly

### ISSUE THREAD

**Issue by winternet-studio** [ForkIssues/TrackerWin#878]
The new feature of stashing single or individual files doesn't seem to work correctly. I right-clicked on the file Controller/TempFakerController.php and chose Stash 1 File. But when I look at the stash it has stashed all the files:

![bug-all-files-stashed](https://user-images.githubusercontent.com/5200270/87932732-49c37580-ca8c-11ea-894d-cb1a0bb73260.png)

**Comment by DanPristupov** [comment]
What should I do to reproduce the problem?

**Comment by winternet-studio** [comment]
I'm not sure what else to say than the right-clicking on one file and select Stash 1 File...

I'm using the latest version, I'm on Windows 10 (not the latest version 2004 though)...

**Comment by DanPristupov** [comment]
Can you show the output of the stash command from the activity manager please?

Here's how it looks in my case:

<img width="908" alt="Screen Shot 2020-07-20 at 14 25 30" src="https://user-images.githubusercontent.com/618115/87937364-e0943000-ca94-11ea-9793-e4cf6fd07305.png">

**Comment by winternet-studio** [comment]
This is the command that eg. is used `git stash push -- "Plugin/PluginName/Model/Behavior/LogableBehavior.php"` - but now I see that it stashes this particular file **PLUS** any **staged** files. **Unstaged** files are not included in the stash.

But maybe there is no way to not have staged files included in the stash? In that case it would probably be good to clarify that in the dialog box (maybe not force checking their checkboxes) - otherwise I can imaging other people will also submit issues like this.

**Comment by DanPristupov** [comment]
> This is the command that eg. is used `git stash push -- "Plugin/PluginName/Model/Behavior/LogableBehavior.php"` 

> Now I see that it stashes this particular file **PLUS** any **staged** files. **Unstaged** files are not included in the stash.

Can you check this in command line? This command only stages the selected file and doesn't touch other files (staged and not staged).

**Comment by winternet-studio** [comment]
This is the initial state of the repo:

```
$ git status
On branch fb_393_custom_email_types
Your branch is ahead of 'origin/fb_393_custom_email_types' by 3 commits.
  (use "git push" to publish your local commits)

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   Model/Address.php
        modified:   Model/EmailAddress.php

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   Controller/TempFakerController.php
        modified:   Plugin/PluginName/Model/Behavior/LogableBehavior.php
```

Then I do the stash:

```
$ git stash push -- "Plugin/PluginName/Model/Behavior/LogableBehavior.php"
Saved working directory and index state WIP on fb_393_custom_email_types: d124b62f WIP Mainly refactoring existing code and added email type flags
```

My stash then looks like this - where you see the staged files Address.php and EmailAddress.php are included even though they were not mentioned in the command:
```
$ git stash show -p
diff --git a/Model/Address.php b/Model/Address.php
index cf8607b8..5eeb084a 100644
--- a/Model/Address.php
+++ b/Model/Address.php
@@ -18,6 +18,7 @@ class Address extends AppModel
[...]
diff --git a/Model/EmailAddress.php b/Model/EmailAddress.php
index 4c4e13ff..7910dc03 100644
--- a/Model/EmailAddress.php
+++ b/Model/EmailAddress.php
@@ -7,6 +7,17 @@ class EmailAddress extends AppModel {
[...]
diff --git a/Plugin/PluginName/Model/Behavior/LogableBehavior.php b/Plugin/PluginName/Model/Behavior/LogableBehavior.php
index f15a4534..9ab97b26 100644
--- a/Plugin/PluginName/Model/Behavior/LogableBehavior.php
+++ b/Plugin/PluginName/Model/Behavior/LogableBehavior.php
@@ -255,6 +255,9 @@ class LogableBehavior extends ModelBehavior
[...]
```

But I guess this is a git issue then?!

The git status is now as follows (the two staged files are still staged):

```
$ git status
On branch fb_393_custom_email_types
Your branch is ahead of 'origin/fb_393_custom_email_types' by 3 commits.
  (use "git push" to publish your local commits)

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   Model/Address.php
        modified:   Model/EmailAddress.php

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   Controller/TempFakerController.php
```

**Comment by DanPristupov** [comment]
Could you show the output of `git version` command as well please?

**Comment by winternet-studio** [comment]
Git version: `git version 2.26.2.windows.1`

Not sure if things mentioned here could have any relevancy: https://stackoverflow.com/questions/14759748/stashing-only-staged-changes-in-git-is-it-possible (I was just stashing a single unstaged file though)

**Comment by DanPristupov** [comment]
I'm very surprised but I confirm this. I reproduced the problem in git 2.23, 2.24.2 and 2.26.2.

Not sure if I missed something in docs or it's a bug.

**Comment by DanPristupov** [comment]
https://www.git-scm.com/docs/git-stash/2.26.0#Documentation/git-stash.txt-ltpathspecgt82308203
> pathspec
> This option is only valid for push command.
> 
> **The new stash entry records the modified states only for the files that match the pathspec.** The index entries and working tree files are then rolled back to the state in HEAD only for these files, too, leaving files that do not match the pathspec intact.

This looks like a bug right?

**Comment by winternet-studio** [comment]
I would say so... but yes, I would also be surprised if we have found a git bug here... seems like a common scenario... That's why I'm wondering whether this maybe is by design or because of an internal restriction in git...!

**Comment by sm-g** [comment]
Same issue when stash selected files during merging conflicts after rebase.

It this feature "does not work" at during merging conflicts, the it is better to disable menu item

**Comment by hawkerm** [comment]
Yup, I've been noticing this as well, thought I was going crazy... 

I'm generally trying to stage other files, and then realize I don't want something in the commit, or want to see if I need the change to revert a single file, but want to stash it as a 'backup' or 'undo' ability... so the included staged changes mucking things up (when not selected either) is confusing and makes it harder to re-apply the stash later.
